src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+            for channel in self.l_channels[1:]:
+                type_count.update(channel._determine_apw())
+        return type_count
+
+    @check_normalized
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+
+class BasisSetContainer(NumericalSettings):
+    """
+    A section defining the full basis set used for representing the electronic structure
+    during the diagonalization of a Hamiltonian (component), as defined in `ModelMethod`.
+    This section may contain multiple basis set specifications under the basis_set_components,
+    each with their own parameters.
+    """
+
+    native_tier = Quantity(
+        type=str,  # to be overwritten by a parser `MEnum`
+        description="""
+        Code-specific tag indicating the overall precision based on the basis set parameters.
+        The naming conventions of the same code are used. See the parser implementation for the possible values.
+        The number of tiers varies, but a typical example would be `low`, `medium`, `high`.
+        """,
+    )
+
+    # TODO: add reference to `electronic_structure`,
+    # specifying to which electronic structure representation the basis set is applied
+    # e.g. wavefunction, density, grids for subroutines, etc.
+
+    basis_set_components = SubSection(sub_section=BasisSet.m_def, repeats=True)
+
+    def _determine_apw(self) -> Optional[str]:
+        """
+        Derive the basis set name for a (S)(L)APW case, including local orbitals.
+        Invokes `normalize` on `basis_set_components`.
+        """
+        answer, has_plane_wave = '', False
+        for comp in self.basis_set_components:
+            if isinstance(comp, MuffinTinRegion):
+                comp.normalize(None, None)
+                type_count = comp._determine_apw()
+                if sum([type_count[i] for i in ('apw', 'lapw', 'slapw')]) > 0:
+                    if type_count['slapw'] > 0:
+                        answer += 'slapw'.upper()
+                    elif type_count['lapw'] > 0:
+                        answer += 'lapw'.upper()
+                    elif type_count['apw'] > 0:
+                        answer += 'apw'.upper()
+                    if type_count['lo'] > 0:
+                        answer += '+lo'
+            elif isinstance(comp, PlaneWaveBasisSet):
+                has_plane_wave = True
+        return answer if has_plane_wave else None
+
+    def _smallest_mt(self) -> MuffinTinRegion:
+        """
+        Scan the container for the smallest muffin-tin region.
+        """
+        mt_min = None
+        for comp in self.basis_set_components:
+            if isinstance(comp, MuffinTinRegion):
+                if mt_min is None or comp.radius < mt_min.radius:
+                    mt_min = comp
+        return mt_min
+
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super().normalize(archive, logger)
+        pws = [
```

Comment:
pws, comp... you can use the longer name here

Reply:
... I think the definition made it quite clear in this case, but alright.